test/onnx/onnx_test.cpp

```diff
@@ -2454,9 +2519,387 @@ TEST_CASE(if_pl_test)
+        then_mod->add_instruction(migraphx::make_op("multibroadcast", {{"out_lens", {2, 3}}}), x);
+    auto rt = then_mod->add_instruction(migraphx::make_op("add"), broad_x, l1);
+    then_mod->add_return({rt});
+
+    auto* else_mod = p.create_module("If_5_else");
+    auto re        = else_mod->add_instruction(migraphx::make_op("mul"), y, l2);
+    else_mod->add_return({re});
+
+    auto ret = mm->add_instruction(migraphx::make_op("if"), {cond}, {then_mod, else_mod});
+    auto r   = mm->add_instruction(migraphx::make_op("get_tuple_elem", {{"index", 0}}), ret);
+    mm->add_return({r});
+
+    auto prog = migraphx::parse_onnx("if_then_empty_shape_test.onnx");
+    EXPECT(p == prog);
+}
+
+TEST_CASE(if_then_trailing_one_shape_test)
+{
+    migraphx::program p;
+    auto* mm = p.get_main_module();
+    migraphx::shape sc{migraphx::shape::bool_type, {1}};
+    auto cond = mm->add_literal(migraphx::literal(sc, {1}));
+    migraphx::shape s{migraphx::shape::float_type, {2, 1}};
+    migraphx::shape s_trail{migraphx::shape::float_type, {2}};
+    std::vector<float> ones(s.elements(), 1.0f);
+    auto l1                 = mm->add_literal(s_trail, ones);
+    std::vector<float> rand = {1.24251, 0.23161};
+    auto l2                 = mm->add_literal(s, rand);
+    auto x                  = mm->add_parameter("x", s_trail);
+    auto y                  = mm->add_parameter("y", s);
+
+    auto* then_mod = p.create_module("If_5_if");
+    auto rt        = then_mod->add_instruction(migraphx::make_op("add"), x, l1);
+    auto broad_rt  = then_mod->add_instruction(migraphx::make_op("unsqueeze", {{"axes", {1}}}), rt);
+    then_mod->add_return({broad_rt});
+
+    auto* else_mod = p.create_module("If_5_else");
+    auto re        = else_mod->add_instruction(migraphx::make_op("mul"), y, l2);
+    else_mod->add_return({re});
+
+    auto ret = mm->add_instruction(migraphx::make_op("if"), {cond}, {then_mod, else_mod});
+    auto r   = mm->add_instruction(migraphx::make_op("get_tuple_elem", {{"index", 0}}), ret);
+    mm->add_return({r});
+
+    auto prog = migraphx::parse_onnx("if_then_trailing_one_shape_test.onnx");
+    EXPECT(p == prog);
+}
+
+TEST_CASE(if_then_empty_constant_test)
+{
+    migraphx::program p;
+    auto* mm = p.get_main_module();
+    migraphx::shape sc{migraphx::shape::bool_type, {1}};
+    auto cond = mm->add_literal(migraphx::literal(sc, {1}));
+    migraphx::shape s{migraphx::shape::int64_type, {2, 1}};
+    std::vector<int> rand = {-1, 0};
+    mm->add_literal(migraphx::shape::int64_type);
+    auto l2 = mm->add_literal(s, rand);
+    auto y  = mm->add_parameter("y", s);
+
+    auto* then_mod = p.create_module("If_4_if");
+    auto then_lit  = then_mod->add_literal(migraphx::shape::int64_type);
+    then_mod->add_return({then_lit});
+
+    migraphx::shape gen_shape(migraphx::shape(s.type(), {1}, {0}));
+    auto literal_ins =
+        then_mod->insert_literal((std::prev(then_mod->end())), migraphx::literal(gen_shape, {0}));
```

Comment:
Dont use `insert_instruction` in the tests, just use `add_instruction`. We should be adding the instructions in the correct order.

Reply:
Okay will do.

Reply:
Done